timbermill-local/src/main/java/com/datorama/oss/timbermill/common/disk/SQLJetDiskHandler.java

```diff
@@ -81,9 +90,64 @@ private void init(){
 		return fetchFailedBulks(true);
 	}
 
+	@Override public List<Event> fetchAndDeleteOverflowedEvents() {
+		return fetchAndDeleteOverflowedEvents(true);
+	}
+
+	private List<Event> fetchAndDeleteOverflowedEvents(boolean deleteAfterFetch) {
+		List<Event> allEvents = new ArrayList<>();
+		ISqlJetCursor resultCursor = null;
+		int fetchedCount = 0;
+		List<Event> events;
+
+		try {
+			db.beginTransaction(SqlJetTransactionMode.WRITE);
+			resultCursor = overFlowedEventsTable.lookup(overFlowedEventsTable.getPrimaryKeyIndexName());
+
+			while (fetchedCount < maxFetchedBulksInOneTime && !resultCursor.eof()) {
```

Comment:
i wonder if we need to fetch from db with the cursor, I mean we already fetching not more than _maxFetchedBulksInOneTime_ so there is no risk of getting a lot of events and killing the db.
or is that for the deletion using the cursor?

Reply:
a bit easier for deletion. can do without the cursor and delete in a query but no real difference.